### PR TITLE
add patch for Arrow v8.0.0 - v11.0.0 to add missing `__arrow_ext_class__` method to `BaseExtensionType` class

### PR DESCRIPTION
**Pull request by PetrKralCZ**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/a/Arrow/Arrow-11.0.0-gfbf-2022b.eb [hunk 1]
```diff
@@ -11,7 +11,11 @@ toolchain = {'name': 'gfbf', 'version': '2022b'}
 
 source_urls = ['https://archive.apache.org/dist/%(namelower)s/%(namelower)s-%(version)s']
 sources = ['apache-arrow-%(version)s.tar.gz']
-checksums = ['2dd8f0ea0848a58785628ee3a57675548d509e17213a2f5d72b0d900b43f5430']
+patches = ['Arrow-8.0.0_bugfix.patch']
```
Comment (boegel) [inline easybuild/easyconfigs/a/Arrow/Arrow-11.0.0-gfbf-2022b.eb:14]: @PetrKralCZ The patch filename should be a bit more descriptive, "bugfix" is way too generic.

How about `Arrow-8.0.0_fix-BaseExtensionType-arrow-ext-methods.patch`

File: easybuild/easyconfigs/a/Arrow/Arrow-8.0.0_bugfix.patch [hunk 2]
```diff
@@ -0,0 +1,24 @@
+patch fixes a bug mentioned here: https://github.com/vscentrum/vsc-software-stack/issues/150#issuecomment-1862807311
+the bug is fixed in Arrow version 14: https://github.com/apache/arrow/pull/33802
```
Comment (boegel) [inline easybuild/easyconfigs/a/Arrow/Arrow-8.0.0_bugfix.patch:2]: mention error explicitly:

```suggestion
fix for:
AttributeError: 'pyarrow.lib.BaseExtensionType' object has no attribute '__arrow_ext_class__'
see https://github.com/apache/arrow/pull/33802
```

keep in mind that changes the patch description implies updating the SHA256 checksum

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by PetrKralCZ** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 3 out of 4 (4 easyconfigs in total)
node3157.skitty.os - Linux RHEL 8.8, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/boegel/5253922be81da89b7f43781946c6747d for a full test report.

File: easybuild/easyconfigs/a/Arrow/Arrow-11.0.0-gfbf-2022b.eb [hunk 3]
```diff
@@ -11,7 +11,12 @@ toolchain = {'name': 'gfbf', 'version': '2022b'}
 
 source_urls = ['https://archive.apache.org/dist/%(namelower)s/%(namelower)s-%(version)s']
 sources = ['apache-arrow-%(version)s.tar.gz']
-checksums = ['2dd8f0ea0848a58785628ee3a57675548d509e17213a2f5d72b0d900b43f5430']
+patches = ['Arrow-8.0.0_fix-BaseExtensionType-arrow-ext-methods.patch']
+checksums = [
+    {'apache-arrow-8.0.0.tar.gz': 'ad9a05705117c989c116bae9ac70492fe015050e1b80fb0e38fde4b5d863aaa3'},
```
Comment (boegel) [inline easybuild/easyconfigs/a/Arrow/Arrow-11.0.0-gfbf-2022b.eb:16]: @PetrKralCZ This should be fixed, for `11.0.0`

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by boegel** [comment]
@boegelbot please test @ jsc-zen2

**Comment by boegel** [comment]
@boegelbot please test @ generoso
EB_ARGS="Arrow-11.0.0-gfbf-2022b.eb"

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3157.skitty.os - Linux RHEL 8.8, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/boegel/ad22a2d62ab34ceecf1e4bd90d570767 for a full test report.

**Review by boegel — APPROVED** [review]
lgtm

**Pull request merged**